acme/acme/messages.py

```diff
@@ -223,21 +213,9 @@ class Status(_Constant):
 STATUS_DEACTIVATED = Status('deactivated')
 
 
-class HasResourceType(Protocol):
-    """
-    Represents a class with a resource_type class parameter of type string.
-    """
-    resource_type: str = NotImplemented
-
-
-GenericHasResourceType = TypeVar("GenericHasResourceType", bound=HasResourceType)
-
-
 class Directory(jose.JSONDeSerializable):
     """Directory."""
 
-    _REGISTERED_TYPES: Dict[str, Type[HasResourceType]] = {}
-
     class Meta(jose.JSONObjectWithFields):
         """Directory Meta."""
         _terms_of_service: str = jose.field('terms-of-service', omitempty=True)
```

Comment:
Just noticed this additional thing we can get rid of, no more `terms-of-service` or its extra case below needed.

Reply:
Last time I tried to get rid of it, everything kept blowing up 😭😭😭 but it seems to have worked this time. Hooray!